pallets/parachain-system/src/lib.rs

```diff
@@ -282,6 +295,40 @@ pub mod pallet {
 					.map(|(recipient, data)| OutboundHrmpMessage { recipient, data })
 					.collect::<Vec<_>>();
 
+			if MaxUnincludedLen::<T>::get().filter(|len| !len.is_zero()).is_some() {
+				let dmp_remaining_messages = PendingDownwardMessages::<T>::get().len() as u32;
+				let limits =
+					TotalBandwidthLimits::new(&relevant_messaging_state, dmp_remaining_messages);
+
+				let hrmp_outgoing = outbound_messages
+					.iter()
+					.map(|msg| {
+						(
+							msg.recipient,
+							HrmpChannelUpdate { msg_count: 1, total_bytes: msg.data.len() as u32 },
+						)
+					})
+					.collect();
+				let dmp_processed_count = ProcessedDownwardMessages::<T>::get();
+				let used_bandwidth = UsedBandwidth {
+					ump_msg_count,
+					ump_total_bytes,
+					hrmp_outgoing,
+					dmp_processed_count,
+				};
+				// The bandwidth constructed was ensured to satisfy relay chain constraints.
+				let ancestor = Ancestor::new_unchecked(used_bandwidth, todo!());
```

Comment:
We need a head data of this block, but it's only available during the call to `fetch_collation_info` runtime api (because the value itself is produced by the consensus implementation on the node side).
If I attempt to store it, changes would be discarded anyway.

Reply:
cc @bkchr

Reply:
Yeah, it's impossible to get that value at this stage. What should be done is to leave a "pending" value which is then updated with `system::parent_hash()` during `on_initialize` of the next block.

Reply:
[8f246c9](https://github.com/paritytech/cumulus/pull/2438/commits/8f246c9439fb223a370706cad71c3d60ff87eb17)